base/strings/string.jl

```diff
@@ -231,20 +231,17 @@ isvalid(s::String, i::Integer) =
 
 function getindex(s::String, r::UnitRange{Int})
     isempty(r) && return ""
-    i, j = first(r), last(r)
     l = sizeof(s)
-    if i < 1 || i > l
-        throw(BoundsError(s, i))
-    end
+    i = first(r)
+    (i < 1 || i > l) && throw(BoundsError(s, i))
     @inbounds si = codeunit(s, i)
-    if is_valid_continuation(si)
-        throw(UnicodeError(UTF_ERR_INVALID_INDEX, i, si))
```

Comment:
Better continue throwing a `UnicodeError` rather than a `BoundsError`, as the former gives more details about the problematic code point. This can be helpful if you import invalid Unicode data and want to understand what's the problem.